### DISCUSSION TITLE
Is it possible to read a file from the file system?

### DISCUSSION THREAD

**Discussion by benjamin-thomas** [gleam-lang/gleam#1699] [Questions & support] [answered]
I had a quick look at stdlib but found nothing.

Thanks :)

**Comment by htstudios** [comment]
I'd suggest joining our community discord for questions: https://discord.gg/twY7ZhKTM3

If you are using Erlang as the server side VM then there is https://github.com/gleam-lang/erlang

**Comment by lpil** [comment, marked as answer]
Check out this module in the `gleam_erlang` package

https://hexdocs.pm/gleam_erlang/gleam/erlang/file.html

**Reply by benjamin-thomas** [reply]
Thanks you both @htstudios @lpil :)

Here is a little working example for any future googlers (I'm not sure how to do proper error handling obviously but that's a good start)

Gleam looks (very) nice, I'm very much intrigued by it. I'm looking at the Elixir/Phoenix combo this week and haven't looked at Erlang before. 

My plan would be to hopefully use Gleam for the complicated business logic stuff if it all works out.

Keep up the good work ;)

P.S. @htstudios the discord link does not work. When I click on the  button, I get a reply saying "impossible to accept the invitation"

#### `gleam.toml`
```toml
name = "dictionary"
version = "0.1.0"

[dependencies]
gleam_stdlib = "~> 0.22"
gleam_erlang = "~> 0.10"

[dev-dependencies]
gleeunit = "~> 0.6"
```

####  `dictionary.gleam`
```gleam
import gleam/io
import gleam/erlang/file
import gleam/string
import gleam/list

// Call to Erlang
pub external fn random_int(n) -> Int =
  "rand" "uniform"

fn sys_dict() -> String {
  "/usr/share/dict/words"
}

pub fn random_word() -> String {
  assert Ok(contents) = file.read(sys_dict())
  let words = string.split(contents, on: "\n")
  let rand_idx = random_int(list.length(words) - 1)
  assert Ok(word) = list.at(words, rand_idx)
  word
}

pub fn main() {
  io.println(random_word())
}
```

**Comment by htstudios** [comment]
@benjamin-thomas sorry for linking to gh instead of hex.pm latter is usually much more helpful.

AFAIR the reason the stdlib does not contain file utilities is platform differences between Erlang and JavaScript.

But for `random()` there is support :) https://hexdocs.pm/gleam_stdlib/gleam/int.html#random and https://hexdocs.pm/gleam_stdlib/gleam/float.html#random

**Reply by benjamin-thomas** [reply]
Nothing to be sorry about @htstudios ;)

Thanks for the heads up about ‘random()’!

**Comment by m4saurabh** [comment]
gleam/erlang/file does not exists how to read file

**Comment by GearsDatapacks** [comment]
It's been removed. Use https://hexdocs.pm/simplifile
